src/common/enum_class_param.h

```diff
@@ -0,0 +1,49 @@
+/*!
+ * Copyright 2015-2018 by Contributors
+ * \file enum_class_param.h
+ * \brief macro for using C++11 enum class as DMLC parameter
+ * \author Hyunsu Philip Cho
+ */
+
+#ifndef XGBOOST_COMMON_ENUM_CLASS_PARAM_H_
+#define XGBOOST_COMMON_ENUM_CLASS_PARAM_H_
+
+#include <dmlc/parameter.h>
+#include <string>
+#include <type_traits>
+
+// specialization of FieldEntry for enum class (backed by int)
+#define DECLARE_FIELD_ENUM_CLASS(EnumClass) \
```

Comment:
Not a request of change, just a random comment.

Reply:
@trivialfis Here, we need macros to define template specialization of `FieldEntry` for the enum class `EnumClass`. Originally, I tried creating a new class with the content of the macro and make subclasses of it, but it didn't work because the methods `add_enum`, `set_default`, and `Init` are not virtual.

Reply:
I added a docstring for `DECLARE_FIELD_ENUM_CLASS` with an example.

Reply:
Nice.